spec/cell_spec.rb

```diff
@@ -45,4 +45,50 @@
     expect(@cell.fired_upon?).to be true
   end
 
+  it 'has a . if its unhit and empty' do
+    cell = Cell.new('B4')
+    expect(cell.render).to eq('.')
+  end
+
+  it 'has an M for a miss if empty' do
+    cell = Cell.new('B4')
+    expect(cell.render).to eq('.')
+    cell.fire_upon
+    expect(cell.render).to eq('M')
+  end
+
+  it 'has a . if unhit and with a ship' do
+    cell = Cell.new('B4')
+    cruiser = Ship.new('cruiser', 3)
+    cell.place_ship(cruiser)
+
+    expect(cell.render).to eq('.')
+  end
+
+  it 'has S if unhit with a ship and the reveal code' do
+    cell = Cell.new('B4')
+    cruiser = Ship.new('cruiser', 3)
+    cell.place_ship(cruiser)
+
+    expect(cell.render(true)).to eq('S')
+  end
+
+  it 'has H if hit and has ship' do
+    cell = Cell.new('B4')
+    cruiser = Ship.new('cruiser', 3)
+    cell.place_ship(cruiser)
+    cell.fire_upon
+    expect(cell.render(true)).to eq('H')
+  end
+
+  it 'shows an X for a sunken ship' do
+    cell = Cell.new('B4')
+    cruiser = Ship.new('cruiser', 3)
+    cell.place_ship(cruiser)
+    cell.fire_upon
+    cruiser.hit
+    expect(cell.render(true)).to eq('H')
+    cruiser.hit
+    expect(cell.render(true)).to eq('X')
+  end
```

Comment:
I would change unhit to not hit in the test. And also we are using an each block so we dont need these cells and cruiser created in the test.

Reply:
AHHH UR RIGHT! on the each blocks! let me fix this before the merge!